clients/client/src/main/java/org/projectnessie/client/http/HttpRequest.java

```diff
@@ -140,8 +142,21 @@ private HttpResponse executeRequest(Method method, Object body) throws HttpClien
           }
         }
         con.connect();
-        con.getResponseCode(); // call to ensure http request is complete
-
+        con.getResponseCode();
+        if (con.getResponseCode() == 308 || con.getResponseCode() == 307) {
```

Comment:
Let's restrict to 308, because 307 is already handled by `HttpURLConnection`. Also use a new enum value in `Status` for 308.